### PR TITLE
L-12 IO Java

### PR DESCRIPTION
**Pull request by IvashchenkoIvanIgorevich**

### CODE REVIEW

File: src/Main.java [hunk 1]
```diff
@@ -1,20 +1,61 @@
...
+                System.out.println("It seems like you haven't entered anything. Bye!!!");
+                scanner.close();
+                return;
+            }
+            System.out.println("You have entered: " + bookTitle + ". " +
+                    "The result of the analysis will be saved in a file with the exact same name.");
+        }
+
+        // find book location
+        Path bookLocated = Paths.get("src", bookTitle);
```
Comment (shaitanaleksander) [inline src/Main.java]: bookLocation

File: src/Main.java [hunk 2]
```diff
@@ -1,20 +1,61 @@
...
+        // create an analysis strategy for getting top and unique words
+        PopularTOPWordStrategy analysis = new PopularTOPWordStrategy(10);
```
Comment (shaitanaleksander) [inline src/Main.java]: class name is confusing

File: src/Main.java [hunk 3]
```diff
@@ -1,20 +1,61 @@
...
+        // main parser which can implement different analysis strategy
+        Parser parser = new Parser(analysis);
+
+        try (Stream<String> lines = Files.lines(bookLocated)) {
+            // parse every line and analyze it
+            parser.pars(lines);
+        } catch (NoSuchFileException e) {
+            System.out.println("The file named " + bookTitle + " could not be located.");
```
Comment (shaitanaleksander) [inline src/Main.java:42]: you need to do something with this exception, just printing message is not good

File: src/homework/io/AnalysisStrategy.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+/**
+ * This interface represents creating new analysis strategy manipulating with line
+ *
+ * https://refactoring.guru/design-patterns/strategy
+ */
+public interface AnalysisStrategy {
+
+    void apply(Stream<String> line);
+}
```
Comment (shaitanaleksander) [inline src/homework/io/AnalysisStrategy.java:13]: you don't need this interface

File: src/homework/io/PopularTOPWordStrategy.java [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+                .limit(top)
+                .map(w -> new StringBuilder()
+                        .append(w.getKey())
+                        .append(" -> ")
+                        .append(w.getValue()))
+                .collect(Collectors.joining("\n"));
+    }
+
+    @Override
+    public void apply(Stream<String> lines) {
```
Comment (shaitanaleksander) [inline src/homework/io/PopularTOPWordStrategy.java]: name is bad

File: src/homework/io/AnalysisStrategy.java (hunk 4 above)
Comment (IvashchenkoIvanIgorevich) [reply src/homework/io/AnalysisStrategy.java:13]: this interface will be used to extend the functionality

**Review by shaitanaleksander — APPROVED** [review]

**Pull request merged**